File: src/com/l2kt/gameserver/network/serverpackets/GMViewWarehouseWithdrawList.java
package com.l2kt.gameserver.network.serverpackets;

import java.util.Set;

import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.model.item.instance.ItemInstance;
import com.l2kt.gameserver.model.item.kind.Item;
import com.l2kt.gameserver.model.item.kind.Weapon;
import com.l2kt.gameserver.model.pledge.Clan;

public class GMViewWarehouseWithdrawList extends L2GameServerPacket
{
	private final Set<ItemInstance> _items;
	private final String _playerName;
	private final int _money;
	
	public GMViewWarehouseWithdrawList(Player player)
	{
		_items = player.getWarehouse().getItems();
		_playerName = player.getName();
		_money = player.getWarehouse().getAdena();
	}
	
	public GMViewWarehouseWithdrawList(Clan clan)
	{
		_playerName = clan.getLeaderName();
		_items = clan.getWarehouse().getItems();
		_money = clan.getWarehouse().getAdena();
	}
	
	@Override
	protected final void writeImpl()
	{
		writeC(0x95);
		writeS(_playerName);
		writeD(_money);
		writeH(_items.size());
		
		for (ItemInstance temp : _items)
		{
			Item item = temp.getItem();
			
			writeH(item.getType1());
			writeD(temp.getObjectId());
			writeD(temp.getItemId());
			writeD(temp.getCount());
			writeH(item.getType2());
			writeH(temp.getCustomType1());
			writeD(item.getBodyPart());
			writeH(temp.getEnchantLevel());
			writeH(temp.isWeapon() ? ((Weapon) item).getSoulShotCount() : 0x00);
			writeH(temp.isWeapon() ? ((Weapon) item).getSpiritShotCount() : 0x00);
			writeD(temp.getObjectId());
			writeD((temp.isWeapon() && temp.isAugmented()) ? 0x0000FFFF & temp.getAugmentation().getAugmentationId() : 0);
			writeD((temp.isWeapon() && temp.isAugmented()) ? temp.getAugmentation().getAugmentationId() >> 16 : 0);
		}
	}
}

File: src/com/l2kt/gameserver/network/serverpackets/GMViewWarehouseWithdrawList.kt
package com.l2kt.gameserver.network.serverpackets

import com.l2kt.gameserver.model.actor.instance.Player
import com.l2kt.gameserver.model.item.instance.ItemInstance
import com.l2kt.gameserver.model.item.kind.Weapon
import com.l2kt.gameserver.model.pledge.Clan

class GMViewWarehouseWithdrawList : L2GameServerPacket {
    private val _items: Set<ItemInstance>
    private val _playerName: String
    private val _money: Int

    constructor(player: Player) {
        _items = player.warehouse.items
        _playerName = player.name
        _money = player.warehouse.adena
    }

    constructor(clan: Clan) {
        _playerName = clan.leaderName
        _items = clan.warehouse.items
        _money = clan.warehouse.adena
    }

    override fun writeImpl() {
        writeC(0x95)
        writeS(_playerName)
        writeD(_money)
        writeH(_items.size)

        for (temp in _items) {
            val item = temp.item

            writeH(item.type1)
            writeD(temp.objectId)
            writeD(temp.itemId)
            writeD(temp.count)
            writeH(item.type2)
            writeH(temp.customType1)
            writeD(item.bodyPart)
            writeH(temp.enchantLevel)
            writeH(if (temp.isWeapon) (item as Weapon).soulShotCount else 0x00)
            writeH(if (temp.isWeapon) (item as Weapon).spiritShotCount else 0x00)
            writeD(temp.objectId)
            writeD(if (temp.isWeapon && temp.isAugmented) 0x0000FFFF and temp.getAugmentation()!!.getAugmentationId() else 0)
            writeD(if (temp.isWeapon && temp.isAugmented) temp.getAugmentation()!!.getAugmentationId() shr 16 else 0)
        }
    }
}